### DISCUSSION TITLE
Multi-threading and expect issues

### DISCUSSION THREAD

**Discussion by FabianBallar7** [QuantumBFS/Yao.jl#519] [Q&A] [answered]
Hey everyone! 

I have a question regarding a weird issue I encountered while trying to run several realizations of a particular circuit with different initial states in parallel. I've managed to pinpoint that the issue is coming from the expect() calls so here's a simplified example which still gives me the same error. 

```julia
using Base.Threads

println(nthreads())

using LinearAlgebra, Yao, Yao.AD, Yao.EasyBuild

N=5

h = heisenberg(N)

Threads.@threads for i = 1:50
    i_state = rand_state(N)
    Ei = expect(h,i_state)
end
``` 

I get a very large error message stating:

Invalid instruction at 0x7f2588427b08: 0xc5, 0xfb, 0x92, 0xc8, 0x62, 0xf1, 0xff, 0x08, 0xc2, 0xd0, 0x07, 0x62, 0xf1, 0xff, 0x09

[1189069] signal (4.2): Illegal instruction
 
 [More error message]
 
Allocations: 4133851 (Pool: 4132253; Big: 1598); GC: 6
Illegal instruction (core dumped)

The weird part is, when I run this same code on my local machine it works just fine. Therefore, I'm unsure if it's just a hardware issue or a matter of permissions. 

I've also tried the same with distributed and the same issue occurs. 

If anyone has any idea of what could be going or experienced something similar I'd appreciate your input!

**Comment by Roger-luo** [comment, marked as answer]
This sounds like a Julia bug but I'm not sure. What's your Julia version?

**Reply by FabianBallar7** [reply]
Hi Roger, thank you for your response.

You're right, this appears to be a Julia bug. I have now updated my Julia to version 1.10.4 and everything seems to be running as intended. Thanks for your input!
